CMakeLists.txt

```diff
@@ -73,17 +75,70 @@ else() #If ROCBLAS not found
   message(FATAL_ERROR "rocblas not found !!! Install rocblas to proceed ...")
 endif(rocblas_FOUND)
 
+find_package (hipblaslt)
+if(hipblaslt_FOUND)
+  message(STATUS "hipblaslt found")
+  set(HIPBLASLT_INC_DIR ${hipblaslt_INCLUDE_DIR} )
+  get_filename_component(HIPBLASLT_LIB_DIR "${HIPBLASLT_INC_DIR}/../lib" ABSOLUTE)
+else() #If HIPBLASLT not found
+  message(FATAL_ERROR "hipblaslt not found !!! Install hipblaslt to proceed ...")
+endif(hipblaslt_FOUND)
+
+find_package (hsa-runtime64)
+if(hsa-runtime64_FOUND)
+  message(STATUS "hsa-runtime64 found")
+  get_target_property(HSA_INC_DIR hsa-runtime64::hsa-runtime64 INTERFACE_INCLUDE_DIRECTORIES)
+  get_filename_component(HSA_PATH "${HSA_INC_DIR}/.." ABSOLUTE)
+  get_filename_component(HSA_LIB_DIR "${HSA_PATH}/lib" ABSOLUTE)
+  set(ROCR_INC_DIR ${HSA_INC_DIR})
+  set(ROCR_LIB_DIR ${HSA_LIB_DIR})
+else() #If HSA not found
+  message(FATAL_ERROR "hsa not found !!! Install hsa to proceed ...")
+endif(hsa-runtime64_FOUND)
+
+find_package (hip)
+if(hip_FOUND)
+  set(HIP_INC_DIR ${hip_INCLUDE_DIR})
+  get_filename_component(HIP_PATH "${hip_INCLUDE_DIR}/.." ABSOLUTE)
+  get_filename_component(HIP_LIB_DIR "${hip_INCLUDE_DIR}/../lib" ABSOLUTE)
+else() #If HIP not found
+  message(FATAL_ERROR "hip not found !!! Install hip to proceed ...")
+endif(hip_FOUND)
+
+find_package (rocm_smi)
+if(rocm_smi_FOUND)
+  message(STATUS "rocm_smi found")
+  set(ROCM_SMI_INC_DIR ${rocm_smi_INCLUDE_DIR} )
+  set(ROCM_SMI_LIB_DIR ${rocm_smi_LIB_DIR} )
```

Comment:
why is get_filename_comnponent not needed here to get absolute path of smi directory?

Reply:
rocm_smi_INCLUDE_DIR and rocm_smi_LIB_DIR are already set in rocm_smi-config.cmake with below cmake.in
https://github.com/ROCm/rocm_smi_lib/blob/amd-staging/rocm_smi-config.cmake.in#L10